packages/hmr-plugin/src/hmr-manager.ts

```diff
@@ -53,20 +62,54 @@ export class HmrManager<T extends Partial<NgxsHmrLifeCycle<S>>, S = NgxsHmrSnaps
   }
 
   public createNewModule(): void {
-    const appRef: ApplicationRef = this.ngModule.injector.get(ApplicationRef);
-    const elements: ComponentRef<any>[] = appRef.components.map(c => c.location.nativeElement);
-    const createNewModule: () => void = createNewHosts(elements);
-    removeNgStyles();
+    const removeOldHosts: () => void = this.cloneHostsBeforeDestroy();
+    this.removeNgStyles();
     this.ngModule.destroy();
-    createNewModule();
+    removeOldHosts();
   }
 
   private createLifecycle(): HmrLifecycle<T, S> {
-    const ngAppModule = this.ngModule.instance;
-    const storage = this.storage;
-    const context = this.context;
-    const bootstrap = this.ngModule.injector.get(NgxsBootstrapper);
-    const builder = this.optionsBuilder;
-    return new HmrLifecycle(ngAppModule, bootstrap, storage, context, builder);
+    return new HmrLifecycle(
+      this.ngModule.instance,
+      this.bootstrap,
+      this.storage,
+      this.context,
+      this.optionsBuilder
+    );
+  }
+
+  private cloneHostsBeforeDestroy(): () => void {
+    const elements: Element[] = this.applicationRef.components.map(
+      (component: ComponentRef<Element>) => component.location.nativeElement
+    );
+
+    const removableList: OldHostRemoverFn[] = elements.map((componentNode: Element) => {
```

Comment:
Do we really need `OldHostRemoverFn` type? In theory TS compiler infers type from the return value itself

Reply:
Although I agree that it is not entirely necessary, it carries value in terms of self documenting code to communicate the intent of the construct. I think it should remain.